packages/types/src/primitive/Event.ts

```diff
@@ -126,8 +150,46 @@ export default class Event extends Struct {
     };
   }
 
+  static getDataType (soughtEventIndex: Uint8Array, metadata: Metadata): Constructor<EventData> | undefined {
+    const methods = ([] as {meta: EventMetadataV5, eventIndex: Uint8Array, sectionName: string}[]).concat(
```

Comment:
2 things:
- the casting is wrong imo: `metadata.asV5.modules` is `ModuleMetadata[]`, so should be the empty array
- small formatting nit: .concat could go on a new line

Reply:
> the casting is wrong imo: metadata.asV5.modules is ModuleMetadata[], so should be the empty array

I'm mapping `metadata.asV5.modules` so that it becomes an array of arrays of `{meta, eventIndex, sectionName}` ; and then I'm using `[].concat(...arraysOfArrays)` to flatten. The end array is an array of `{meta, eventIndex, sectionName}`, so I don't think the casting is wrong?

Reply:
Ah my bad. I was reading `([] as ...).concat(...metadata.asV5.modules).filter(...).map(...)`.

In this case the .concat formatting is better like this.